### PR TITLE
Update MklMatMulPrimitiveFactory to support Arm Compute Library backend

### PR DESCRIPTION
**Pull request by cfRod**
Related to issue #47415 and PR #47775. Adding support for caching matmul primitives.
Updates onednn_acl_primitives.patch to include matmul primitives.

### CODE REVIEW

File: tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h [hunk 1]
```diff
@@ -697,7 +700,9 @@ class MklMatMulPrimitiveFactory : public MklPrimitiveFactory<T> {
     key_creator.AddAsKey(params.b_strides);
     key_creator.AddAsKey(params.c_strides);
     key_creator.AddAsKey(typeid(T).name());
-
+#ifdef DNNL_AARCH64_USE_ACL
+  key_creator.AddAsKey(params.randn);
```
Comment (penpornk) [inline tensorflow/core/kernels/mkl/mkl_matmul_ops_common.h:704]: Nit: Indentation seems to be off.

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for the PR!

File: tensorflow/core/kernels/mkl/mkl_batch_matmul_op.cc [hunk 2]
```diff
@@ -140,6 +144,14 @@ class BatchMatMulMkl : public OpKernel {
     MklBatchMatMulHelper bmm;
     auto params = bmm.CreateMatMulParams(lhs.shape(), rhs.shape(), out_shape,
                                          adj_x_, adj_y_);
+
+#ifdef DNNL_AARCH64_USE_ACL
+    // ACL does not support reuse of primitives with different data.
+    // For matmul, the previous approach (PR #47775) of using Tensor addresses
+    // does not work, as the addresses are re-used in matmul with different data
+    // Use of rand() ensures we still benefit from caching via SetMklMatmul().
+    params->randn = rand();
```
[the remark this answers is not in the document]
Comment (cfRod) [reply tensorflow/core/kernels/mkl/mkl_batch_matmul_op.cc:153]: Thanks I will resolve this!

**Review by penpornk — APPROVED** [review]
Thank you for the changes! :)

**Comment by cfRod** [comment]
@penpornk thanks!
If this can be cherry-picked onto 2.7 rls that would be great!

**Pull request merged**